lib/assert.js

```diff
@@ -280,9 +282,89 @@ function _deepEqual(actual, expected, strict, memos) {
   memos.actual.push(actual);
   memos.expected.push(expected);
 
+
   return objEquiv(actual, expected, strict, memos);
 }
 
+function setEquiv(a, b, strict, actualVisitedObjects) {
+  // This code currently returns false for this pair of sets:
+  //   assert.deepEqual(new Set(['1', 1]), new Set([1]))
+  //
+  // In theory, all the items in the first set have a corresponding == value in
+  // the second set, but the sets have different sizes. Its a silly case,
+  // and more evidence that deepStrictEqual should always be preferred over
+  // deepEqual. The implementation currently returns false, which is simpler
+  // and slightly faster.
+  if (a.size !== b.size)
+    return false;
+
+  outer: for (const val1 of a) {
+    if (!b.has(val1)) {
+      if (!strict || (typeof val1 === 'object' && val1 !== null)) {
+        // The value doesn't exist in the second set by reference, so we'll go
+        // hunting for something thats deep-equal to it. Note that this is
+        // O(n^2) complexity, and will get slower if large, very similar sets /
+        // maps are nested inside. Unfortunately there's no real way around
+        // this.
+        for (const val2 of b) {
```

Comment:
If I am reading the code correctly, is it similar to...

```js
for (const val1 of a) {
  if (!b.has(val1)) {
    if (strict) { return false; }
    if (typeof val1 !== 'object' || val1 === null)) { return false; }

   // the following can be wrapped into a funciton...
    var bHasSimilarElement = false;
    for (const val2 of b) {
      if (_deepEqual(val1, val2, strict, actualVisitedObjects)) {
        bHasSimilarElement = true;
        break;
      }
    }

    if (!bHasSimilarElement) {
      return false;
    }
  }
}
```
?

Personally I prefer to stay away from labels...